opencv: Update to 3.1 to fix compilation problems

luaraneda opened the pull request:
This patch is a backport from a recent fix in upstream commit 78376c0f63834642b8133ab5d3b5f658d65a0bb2:
"pthreads: replace usage of non-POSIX PTHREAD_RECURSIVE_MUTEX_INITIALIZER* defines".

It also includes the changes made to the file "modules/core/src/precomp.hpp" in upstream commit: 53fc5440d78fd9ebe727c51243528e1eac3b5e35: "implement singleton lazy initialization".

luaraneda commented:
Updated the pull request to include a new patch that suppress several warnings on GCC 5.x
Tested with GCC 5.2 and 4.8.3.

champtar commented:
@luaraneda sorry for the delay
is this still relevant? (update to opencv 3.1?)
you need to bump PKG_RELEASE

@nakarotori, @lijpsummer if one of you can take a look

luaraneda commented:
I think the commits are no loner relevant since the have been included on the 3.1 release.
I will try to compile the 3.1 release this weekend, and update the pull request accordingly.

luaraneda commented:
I did the rebase, deleted the old commits and added one with just a version update.
This solved the compile problems I had with musl on a sunxi target, but also applies to others targets.

@champtar for what I've read, after an official update (3.1.0), the PKG_RELEASE should be set to 1, so is not necessary to bump it. Is that right?

champtar commented:
@luaraneda you are right, your commit seems OK to me

@nakarotori, @lijpsummer if one of you can take a look
